src/ws/client.rs

```diff
@@ -231,12 +235,12 @@ impl<'a> TryFrom<&'a EspWebSocketClientConfig<'a>> for (esp_websocket_client_con
 
             ping_interval_sec: conf.ping_interval_sec.as_secs() as _,
 
-            cert_pem: cstrs.as_nptr(conf.cert_pem)?,
-            cert_len: conf.cert_pem.map(|c| c.len()).unwrap_or(0) as _,
-            client_cert: cstrs.as_nptr(conf.client_cert)?,
-            client_cert_len: conf.client_cert.map(|c| c.len()).unwrap_or(0) as _,
-            client_key: cstrs.as_nptr(conf.client_key)?,
-            client_key_len: conf.client_key.map(|c| c.len()).unwrap_or(0) as _,
+            cert_pem: cstrs.as_nptr(conf.server_cert.as_ref().map(x509_data))?,
```

Comment:
This is doing an expensive (in terms of RAM usage) copy of the der/pem data which is referenced by the X509 certificate struct into c-strings that are kept internally in the `RawCstrs` struct which is kept around while the client is created, but is then **dropped** by the current code, which is wrong.

I don't think dropping the `RawCstrs` struct is save, as the TLS layer under the WS client *does not do its own copy of the PEM/DER data*. So you have to keep around the `RawCstrs` structure until the client is alive, as its member.
 
I think however it is much better to take another approach: change the types from `X509<'a>` to `X509<'static>` as we do everywhere else. This would enforce the user to pass statically allocated pems and ders, but will remove the RAM consumption and the copying.

And... pems and ders tend to be inbcluded into the flash binary's `.rodata` section using `include_bytes!` / `include_str!` anyway (as you did in your `main` function) so why copy them?

Reply:
The requirement for an explicit null terminator actually makes it challenging to use _include_bytes!()_ and _include_str!()_, as @torkleyy noted when inlining a certificate into the blocking TLS example in 26057471e4698f4ee4162ccfbeb73d3a67a169b7. (I should've added a similar comment to the websocket client example.)

But it isn't impossible, as noted in https://stackoverflow.com/questions/75064278/include-str-for-null-terminated-string. And we can at the very least inline the certs into the source and avoid copying them (a worthy goal, to be sure), as we've done here, by explicitly adding a null terminator.

Reply:
its trivial to add a null termination to a .pem file you get out of for example openssl. For example you can use `dd if=/dev/zero bs=1 count=1 >> your_cert.pem` or a more "safe" way is to use `truncate -s +1 your_cert.pem` and then just include it via the macro.